### PR TITLE
Issue 57

### PR DESCRIPTION
**Pull request by sanjuthomas**
Issue 57 - 
Load json files from directory and its test case. 
Change the java complier version to 8 as per the description in the read me.  
3 JSON files are added into test/data folder for testing.

### CODE REVIEW

File: src/main/java/com/marklogic/spring/batch/job/LoadDocumentsFromDirectoryJob.java [hunk 1]
```diff
@@ -1,43 +1,49 @@
...
+import com.marklogic.uri.DefaultUriGenerator;
 
 @Configuration
 @PropertySource("classpath:job.properties")
 @EnableBatchProcessing
 @Import(com.marklogic.spring.batch.configuration.MarkLogicBatchConfiguration.class)
 public class LoadDocumentsFromDirectoryJob {
 
+    public static final String URI = "uri";
+    private final Log logger = LogFactory.getLog(LoadDocumentsFromDirectoryJob.class);
```
Comment (rjrudin) [inline src/main/java/com/marklogic/spring/batch/job/LoadDocumentsFromDirectoryJob.java]: I recommend extending LoggingObject from ml-javaclient-util. This uses org.slf4j.Logger, and the preferred implementation these days seems to be Logback.

**Comment by rjrudin** [comment]
@sanjuthomas and @sastafford - how would a user of the marklogic-spring-batch (MSB) library make use of this? 

I'm not sure that the MSB library should have any Configuration classes in it. I think it should definitely have a collection of reusable Writer classes, probably some Processor classes (like for doing a transform), and maybe even some Reader classes. A client of MSB would then interact it with it by importing these classes into their own Batch application.

MSB could also offer a command-line interface that would allow a user to select a Job to run and provide any configuration parameters necessary - just like mlcp (which has "jobs" of IMPORT, EXPORT, etc).

**Comment by sastafford** [comment]
I think you misunderstood me about one of my earlier comments.  When I stated I wanted you to modify the LoadDocumentsFromDirectoryJob, I did not mean I wanted you to replace the existing functionality of importing xml documents.  I want this job to recognize the document_type parameter. 

If document_type = xml then load the xml documents
else if document_type = json then load the json documents.  

We want to replicate the same capability as mlcp with this job.  

So this job still needs to be able to load xml documents but if the document_type is json then the ItemReader/ItemWriters need to change based on the document type.   You may need a separate test class to test loading json documents. 

Sorry for the confusion but I need you to merge my original changes and your new changes on this one.

**Comment by sastafford** [comment]
@rjrudin, I think there needs to be reusable ItemReader / ItemWriter classes.  I'm fine with the ItemProcessor being an anonymous class for right now.  For xml we already have a reusable DocumentItemWriter class.  I think we need another one for JSON docs

**Comment by sanjuthomas** [comment]
@sastafford OK. Things were not so clear to me. "Add capability to load JSON as per the document_type param" would have been a clear description". Also, in the acceptance criteria if the test case for xml was also there then it would have pointed me in that direction. I will submit another pull request.

**Pull request closed without merging**